File: tests/pyre.lib/timers/wall_timer_example.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// get the timers
#include <pyre/timers.h>
#include <pyre/journal.h>
// support
#include <thread>
#include <cassert>
// access the {chrono} literals
using namespace std::literals;


// type aliases
using walltimer_t = pyre::timers::wall_timer_t;


// verify that two timers that have the same name share the same movement
int main() {
    // make a timer
    walltimer_t t("tests.timer");
    // start it
    t.start();

    // nap duration
    auto nap = 50ms;
    // go to sleep for a bit
    std::this_thread::sleep_for(nap);

    // stop the timer
    t.stop();

    // make a channel
    pyre::journal::debug_t channel("tests.timer");
    // activate it
    // channel.activate();
    // show me
    channel
        << "elapsed time: " << t.us()
        << pyre::journal::endl(__HERE__);

    // all done
    return 0;
}


// end of file

File: tests/pyre.ext/timers/wall_timer_example.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


def test():
    """
    Exercise the simple use case with a wall clock timer
    """
    # externals
    import time
    # access the timer bindings
    from pyre.extensions.pyre.timers import WallTimer

    # make a timer
    t = WallTimer(name="tests.timer")
    # start it
    t.start()

    # take a nap
    time.sleep(.5)

    # stop the timer
    t.stop()

    # get the journal
    import journal
    # make a channel
    channel = journal.debug(name="pyre.timers")
    # activate it
    # channel.activate()
    # log the elapsed time
    channel.log(f"elapsed: {t.ms()}ms")

    # all done
    return


# main
if __name__ == "__main__":
    # run the test
    test()


# end of file
